### PR TITLE
Add in Jest and React Testing ESLint Plugins

### LINKED ISSUE #33: Add in Testing Best Practice ESLint Plugins
### Issue description
In order to avoid [some common mistakes](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) with React testing, there are a couple of ESLint plugins that can ensure our tests are following best practices. This will allow our tests to be more consistent, and hopefully avoid future hidden issues.

Additionally there is a third testing library around jest to ensure that we use consistent naming in our tests. (Especially `test` vs `it`)

The plugins are:
* [eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library)
* [eslint-plugin-jest-dom](https://github.com/testing-library/eslint-plugin-jest-dom)
* [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest#readme)

> **ESTIMATE** TBD

### What's the expected change?
- Nothing but more consistent tests hopefully

### PR DESCRIPTION
**Pull request by bryanpizzillo**
There will be 4 commits in total, one for turning on Jest linting, one for fixing Jest linting errors, one for turning on react-testing linting, one for fixing the errors.

This closes #33

### CODE REVIEW

**Review by seyilonge-nci — APPROVED** [review]

File: src/components/atomic/audio-player/audio-player.jsx [hunk 1]
```diff
@@ -59,7 +60,9 @@ const AudioPlayer = ({ audioSrc, lang = 'en', tracking = () => {} }) => {
 					paused ? 'paused' : ''
 				}${hasError ? 'error' : ''}`}
 				onClick={handlePlay}>
-				<span className="show-for-sr">{srText}</span>
+				<span data-testid="span.show-for-sr" className="show-for-sr">
```
Comment (arcepaul) [inline src/components/atomic/audio-player/audio-player.jsx]: weird name for a testid

File: src/components/atomic/spinner/__tests__/Spinner.tests.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 import Spinner from '../Spinner';
 
 describe('Spinner component', () => {
-	test('should have expected button label and fire onRemove handler', () => {
-		const { container } = render(<Spinner />);
-		expect(container.querySelector('.nci-spinner')).toBeInTheDocument();
-		expect(container.querySelector('.spinkit')).toBeInTheDocument();
-		expect(container.querySelector('.dot1')).toBeInTheDocument();
-		expect(container.querySelector('.dot2')).toBeInTheDocument();
+	it('should have expected button label and fire onRemove handler', () => {
```
Comment (arcepaul) [inline src/components/atomic/spinner/__tests__/Spinner.tests.js:7]: A new issue was entered to revisit the spinner and its tests. #35 

fine for this PR

File: src/components/atomic/audio-player/audio-player.jsx (hunk 1 above)
Comment (bryanpizzillo) [reply src/components/atomic/audio-player/audio-player.jsx]: I was able to remove test ids from the audio-player and the video-player. Some of the other ones are too embedded.

File: src/components/atomic/InputLabel/__tests__/InputLabel.tests.js [hunk 3]
```diff
@@ -18,11 +18,10 @@ describe('', function () {
 		/>
 	);
 
-	test('InputLabel renders with correct class names and label hint', function () {
-		const { getByTestId, getByText } = wrapper;
-		const label = getByTestId(`tid-${htmlFor}-label`);
+	it('InputLabel renders with correct class names and label hint', function () {
```
Comment (arcepaul) [inline src/components/atomic/InputLabel/__tests__/InputLabel.tests.js:21]: Input Label is redundant here.  I always read the test description as a continuation of the describe text, is that what we should be doing?

Comment (seyilonge-nci) [reply src/components/atomic/InputLabel/__tests__/InputLabel.tests.js:21]: I wrote this one. Agreed, it's redundant here. Would be nice we follow "it('should...')" type language.

**Review by arcepaul — APPROVED** [review]

**Pull request merged**